### PR TITLE
Fix first_image not resolving correctly in subdirectories

### LINKED ISSUE #59: first_image URLs not resolving correctly on RTD
Thanks for this amazing package! 👏  So helpful for the open source documentation community.

## Context

I am trying to use this extension with a sphinx site (https://github.com/pangeo-forge/pangeo-forge-recipes/tree/master/docs) on ReadTheDocs. My config looks like this

```python
ogp_image = "_static/pangeo-forge-logo-blue.png"
ogp_use_first_image = True
```

I am explicitly **not setting** `ocp_site_url`, as I want it to be set automatically, following this code path:

https://github.com/wpilibsuite/sphinxext-opengraph/blob/d99815bac5a42f5e9ac756cc3e74eebfd80d2505/sphinxext/opengraph/__init__.py#L70-L72

As far as I can tell, this is working correctly.

### What works:  default `ogp_image`

Our home page - <https://pangeo-forge.readthedocs.io/en/latest/index.html> - has no image so defaults to `ogp_image`. The headers look like this:

```html
<meta property="og:title" content="Pangeo Forge Documentation" />
<meta property="og:type" content="website" />
<meta property="og:url" content="https://pangeo-forge.readthedocs.io/en/latest/index.html" />
<meta property="og:description" content="Resources for understanding and using Pangeo Forge First Steps: New to Pangeo Forge? You are in the right spot! What is Pangeo Forge?- Read more about Pangeo Forge and how it works., Introduction T..." />
<meta property="og:image" content="https://pangeo-forge.readthedocs.io/en/latest/_static/pangeo-forge-logo-blue.png" />
<meta property="og:image:alt" content="Pangeo Forge Documentation" />
```

The image url - <https://pangeo-forge.readthedocs.io/en/latest/_static/pangeo-forge-logo-blue.png> - is correct and works.

### What does not work: first_image

This page -  <https://pangeo-forge.readthedocs.io/en/latest/introduction_tutorial/intro_tutorial_part1.html> - DOES have an image.

It's header looks like this

```html
<meta property="og:title" content="Defining a FilePattern (Intro Tutorial Part 1)" />
<meta property="og:type" content="website" />
<meta property="og:url" content="https://pangeo-forge.readthedocs.io/en/latest/introduction_tutorial/intro_tutorial_part1.html" />
<meta property="og:description" content="Welcome to the Pangeo Forge introduction tutorial! This tutorial is split into three parts: Defining a FilePattern, Defining a recipe and running it locally, Setting up a recipe to run in the cloud..." />
<meta property="og:image" content="https://pangeo-forge.readthedocs.io/en/_images/OISST_URL_structure.png" />
<meta property="og:image:alt" content="OISST file structure" />
```

Note that here the `og:image` URL is not correct! <https://pangeo-forge.readthedocs.io/en/_images/OISST_URL_structure.png>

It should be <https://pangeo-forge.readthedocs.io/en/latest/_images/OISST_URL_structure.png>

The autogenerated image URL is  missing the `/latest` directory.

#48 was supposed to add support for relative URLs, but something is not working right here. I would appreciate any suggestions you may have.

---

🙏  thanks again for your work on this very useful package!

### PR DESCRIPTION
**Pull request by ItayZiv**
Fixes #59.
I'm actually not 100% if there is a more proper way to do this is, so the solution is a bit hacky, but this will be dealt with in #60 in a more proper way anyways.
(Another solution would be to use `Path(page_url).parent` instead of `ogp_site_url` which would resolve the relative link by itself, but that didn't seem like a cleaner solution, and once again this is a somewhat temporary fix)

### CODE REVIEW

File: sphinxext/opengraph/__init__.py [hunk 1]
```diff
@@ -134,7 +134,9 @@ def get_tags(
             image_url_parsed = urlparse(image_url)
             if not image_url_parsed.scheme:
                 # Relative image path detected. Make absolute.
-                image_url = urljoin(config["ogp_site_url"], image_url_parsed.path)
+                image_url = urljoin(
+                    config["ogp_site_url"], image_url_parsed.path.split("../")[-1]
```
Comment (pradyunsg) [inline sphinxext/opengraph/__init__.py:138]: You can use str.rpartition to do this, which is both more efficient and likely conveys intent more clearly.

**Comment by ItayZiv** [comment]
I actually ended up swapping it over to use `page_url` directly, I forgot that because of how `urljoin` works we don't need to get the parent directory so it ends up looking much cleaner this way.

**Comment by TheTripleV** [comment]
Could you add in a test case that fails before and passes with this PR?

**Comment by ItayZiv** [comment]
Apparently a test for this exists (`test-image-rel-paths`), except its `conf.py` had the url set to `http://example.org/"` which doesn't encounter this problem.
On the other hand, this PR will break relative paths as defined in `conf.py`, which is also untested but while it is mentioned in the `readme` its barley supported since we don't (yet) copy the images, so they have to be placed in `_static` or manually added to `html_static_path` or `html_exclude_path`.
I'll fix the test and I think ill change all URLs to use something like `/en/latest/` since its a *very* common use case, and I can't imagine something which works with it, not working without.
But we should decide whether to axe support for relative paths until #60 for `conf.py` on top of field lists (meaning it will only be useful for `first_image` or fix support (either convert relative paths the old way for `ogp_image`)

**Comment by TheTripleV** [comment]
We shouldn't break existing functionality.
Can you special case the urljoin so that if `ogp_image` is specified, it will still be relative to the `config["ogp_site_url"]`?
This way, both existing functionality (`ogp_image`) and intended functionality (first image) will hold.

**Comment by ItayZiv** [comment]
I fixed it, the fix is a bit weird but I think it is ok until #60.
> This PR doesn't need to add relative path support for field lists.
I meant maybe we should consider not supporting them for now at all, but never mind that.

**Review by TheTripleV — APPROVED** [review]

**Pull request merged**